core/deployment/src/main/java/io/quarkus/deployment/dev/DevModeMain.java

```diff
@@ -134,7 +134,7 @@ public void start() throws Exception {
                     map);
         } catch (Throwable t) {
             log.error("Quarkus dev mode failed to start", t);
-            throw new RuntimeException(t);
+            throw (t instanceof RuntimeException ? (RuntimeException) t : new RuntimeException(t));
```

Comment:
I know it's not exactly pretty but I prefer using this pattern when there is actual code doing something that we would have to keep consistent in both branches.
In this case, it's just a logger so we could discuss it but for other locations, it might be a `close()` or something critical.

Reply:
Personally, I do it the other way, but both are fine I guess.

Another option would be to have a method that either returns its argument (when `instanceof RuntimeException`) or wraps (when not). But the changes here are scattered around so having a common method would likely be more painful than useful.

LGTM, in any case :-)

Reply:
Yeah, I think we can live with it :).